### PR TITLE
Register webhook example

### LINKED ISSUE #50: Finish adding a webhook to the server example
We created the proper hook for after install but we didn't complete the example of how to create a webhook:

https://github.com/Shopify/shopify-node-app/blob/master/server/index.js#L33

We're trying to keep the server code simple so I think we should do this without an api client library and write a plain web request to do this.

https://help.shopify.com/api/reference/webhook

### PR DESCRIPTION
**Pull request by kevinhughes27**
closes #50 

I opted to use an api client library for this example instead because otherwise it was going to be more lines of code than I wanted. I also bumped the minimum node version - was there a reason it wanted version 7 only? I don't have this version on my machine so I updated it.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -30,11 +31,26 @@ const shopifyConfig = {
...
+      format: 'json'
+    });
+
     return response.redirect('/');
   },
 };
 
+const registerWebhook = function(shopDomain, accessToken, webhook) {
+  const shopName = shopDomain.replace('.myshopify.com', '');
+  const shopify = new Shopify({ shopName: shopName, accessToken: accessToken });
```
[the remark this answers is not in the document]
Comment (kevinhughes27) [reply server/index.js]: It does yes. 

If you think about it there will always be 2 different ways to access the Shopify API with this project - calls initiated from the frontend and calls initiated from the backend. We could do some work to make them more similar though such as updating the API proxy to use the client library.

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "precommit": "yarn run pretty"
   },
   "engines": {
-    "node": "~7.7.3"
+    "node": ">= 8.1.0"
```
[the remark this answers is not in the document]
Comment (kevinhughes27) [reply package.json]: Done!

**Comment by kevinhughes27** [comment]
@AWaselnuk we have to balance complexity here. On one hand yes this adds a second way to interact with the Shopify API. On the other hand to make a server side API call to register a webhook without a client library will be fairly verbose. This complexity is hidden behind the API proxy for calls that originate from the react client.

File: server/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheMallen) [reply server/index.js]: How can the api proxy use the client library? It makes arbitrary requests

**Comment by TheMallen** [comment]
Actually, that api client isn't even ours is it? Not sure if we want to include a third party api client.

**Comment by kevinhughes27** [comment]
That is the library that @jamiemtdwyer's team recommends though. 

Its adds a fair amount of complexity to the example to do it manually when the real intent is to show where and how you should execute server side actions after the app is installed. 

Similarly if you add a webhook route its pretty likely that you'll want to callback to Shopify after receiving the webhook which most people would prefer to use a client library for. For example consider my surge pricing example app https://github.com/kevinhughes27/shopify-surge-pricing

**Pull request merged**